src/video/x11/SDL_x11window.c

```diff
@@ -1871,22 +1872,25 @@ X11_FlashWindow(_THIS, SDL_Window * window, SDL_FlashOperation operation)
     return 0;
 }
 
-bool SDL_X11_SetWindowTitle(Display* display, Window xwindow, char* title) {
+int SDL_X11_SetWindowTitle(Display* display, Window xwindow, char* title) {
     Atom _NET_WM_NAME = X11_XInternAtom(display, "_NET_WM_NAME", False);
     XTextProperty titleprop;
-    int conv = X11_XmbTextListToTextProperty(display, (char**) &title, 1, XStdICCTextStyle, &titleprop);
+    int conv = X11_XmbTextListToTextProperty(display, (char**) &title, 1, XTextStyle, &titleprop);
     Status status;
 
+    if (X11_XSupportsLocale() != True) {
+        return SDL_SetError("Current locale not supported by X server, cannot continue.");
+    }
+
     if (conv == 0) {
         X11_XSetTextProperty(display, xwindow, &titleprop, XA_WM_NAME);
         X11_XFree(titleprop.value);
-    // we know this can't be a locale error as we checked X locale validity in X11_VideoInit
-    } else if (conv <= 0) {
-        SDL_LogError(SDL_LOG_CATEGORY_VIDEO, "X11 reporting it's out of memory");
-        return EXIT_FAILURE;
-    } else { // conv >= 0
-        SDL_LogWarn(SDL_LOG_CATEGORY_VIDEO, "%d characters were not convertable to the current locale!", conv);
-        return EXIT_FAILURE;
+    // we know this can't be a locale error as we checked X locale validity
```

Comment:
Minor detail, we use /* */ style comments in SDL, to support really old compilers. :)

Reply:
On it, thanks, sorry for being a n00b, thank you for letting me clean up the mess I made, it's a tad embarrassing given that one of the things my career is based on is text encodings, I'm even a Unicode individual member and attend the meetings whenever I can make it :laughing: I just didn't expect any modern software to fail without `WM_NAME`, and the way this bug was found was such that I don't think any amount of testing would've lead me to find it. :(